### PR TITLE
Add team select search

### PR DESCRIPTION
**Pull request by jafacakes2011**
Adds the team selector updates:
- Matches new designs
- Ordered by name (with options that have children going to the top)

### CODE REVIEW

File: assets/stylesheets/components/_peoplefinder.scss [hunk 1]
```diff
@@ -287,54 +287,60 @@
   color: govuk-colour("red");
 }
 
-// Team select web component styles
-.team-select__teams {
-  margin-bottom: 20px;
-  border: 2px solid black;
+// Team select
+.team-select {
+  font-family: "GDS Transport", arial, sans-serif;
```
Comment (marcelkornblum) [inline assets/stylesheets/components/_peoplefinder.scss:292]: why do we need to add a new font at this level?

File: assets/stylesheets/components/_peoplefinder.scss [hunk 2]
```diff
@@ -287,54 +287,60 @@
...
 
-  ul {
-    list-style: none;
-    margin: 0;
-    padding: 0;
+  &__teams {
+    max-height: 800px;
```
Comment (marcelkornblum) [inline assets/stylesheets/components/_peoplefinder.scss:295]: I'm not sure this is wrong in the context of the rest of the CSS, but it grinds my gears a bit having arbitrary numbers specified in px. Would it not make sense as a relation to either the viewport or the line height?

File: assets/stylesheets/components/_peoplefinder.scss [hunk 3]
```diff
@@ -287,54 +287,60 @@
...
+    overflow-y: auto;
+    margin-bottom: 20px;
```
Comment (marcelkornblum) [inline assets/stylesheets/components/_peoplefinder.scss:297]: same for these measurements - probably not worth updating but don't we have a set of standardised margins etc accessible via scss var?

File: package.json [hunk 4]
```diff
@@ -1,35 +1,38 @@
...
+        "last 1 firefox version",
+        "last 1 safari version",
+        "IE 11"
+    ],
     "engines": {
         "npm": ">8.0.0",
         "node": ">18.0.0"
+    },
+    "prettier": {
+        "printWidth": 100
```
Comment (marcelkornblum) [inline package.json:36]: not sure i agree with just adding in linting overrides ...

**Review by marcelkornblum — APPROVED** [review]
I didn't really review the JS because it was complex to start with, and the linting made it super hard to see what to review... otherwise it's OK though there are a couple of comments :)

File: assets/stylesheets/components/_peoplefinder.scss (hunk 1 above)
Comment (jafacakes2011) [reply assets/stylesheets/components/_peoplefinder.scss:292]: Looks like we don't, I assume initially it was needed as the markup didn't have any gov classes

**Pull request merged**